sdk/impactreporting/arm-impactreporting/src/api/impactContext.ts

```diff
@@ -19,14 +20,17 @@ export interface ImpactClientOptionalParams extends ClientOptions {
   /** The API version to use for this operation. */
   /** Known values of {@link KnownVersions} that the service accepts. */
   apiVersion?: string;
+  /** Specifies the Azure cloud environment for the client. */
+  cloudSetting?: AzureSupportedClouds;
 }
 
 export function createImpact(
   credential: TokenCredential,
   subscriptionId: string,
   options: ImpactClientOptionalParams = {},
 ): ImpactContext {
-  const endpointUrl = options.endpoint ?? options.baseUrl ?? "https://management.azure.com";
+  const endpointUrl =
+    options.endpoint ?? getArmEndpoint(options.cloudSetting) ?? "https://management.azure.com";
```

Comment:
@wanlwanl Could you prioritize the issue? It impacts all Modular service release - https://github.com/Azure/azure-sdk-tools/issues/11825

@v-jiaodi the version in below is wrong, please update it manually. you could refer the version in changelog.

```
const userAgentInfo = `azsdk-js-arm-impactreporting/1.0.0-beta.1`;
```